### PR TITLE
Add RELEASE.md describing the release process and use separate version params in independent pom.xml files

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds new `RELEASE.md` file describing the release process. It also adds new properties for managing the versions of the Strimzi Kafka OAuth libraries.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```
+mvn versions:set -DnewVersion=X.Y.Z
+```
+
+Where `X.Y.Z` is the new version which is going to be released.
+
+## Update dependencies in examples and tests
+
+Update the version in dependencies in:
```
Comment (mstruk) [inline RELEASE.md]: I propose more explicit directions:

Update the `strimzi-oauth.version` property in:

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* `examples/docker/pom.xml`
+* `testsuite/pom.xml`
+
+These are independent Maven projects, so the dependency version needs ot be bumped manually.
```
Comment (mstruk) [inline RELEASE.md]: Type: needs ot -> needs to

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Set new version in main Maven project
+
+Update the version for the new Maven project:
+
+```
+mvn versions:set -DnewVersion=X.Y.Z
+```
+
+Where `X.Y.Z` is the new version which is going to be released.
```
Comment (ppatierno) [inline RELEASE.md]: should be M=X and Y=N? Why not using them, it could be confusing

Comment (scholzj) [reply RELEASE.md]: `X.Y.Z` is the normal placeholder for the version. But in the branch name it conflicets with the `.x` at the end. That is why different letters are used here and there.

**Pull request merged**